### ISSUE TITLE
Inactive `input-message` adds white-space below a `label`

### ISSUE THREAD

**Issue by nwhittaker** [Esri/calcite-components#5130]
### Actual Behavior

Given a `<calcite-label>` that contains a `<calcite-input>` and an inactive `<calcite-input-message>`, its bottom white-space is greater than a `<calcite-label>` that only contains a `<calcite-input>`.

### Expected Behavior

Given a `<calcite-label>` that contains a `<calcite-input>` and an inactive `<calcite-input-message>`, its bottom white-space is the same as a `<calcite-label>` that only contains a `<calcite-input>`.

### Reproduction Sample

https://codepen.io/nwhittaker-esri/pen/QWmBoEe

### Reproduction Steps

1. Visit the code sample
2. Note how the boxes around each `<calcite-label>` are different heights

The height of the 2nd box should match the height of the 1st box.

### Reproduction Version

beta.86

### Relevant Info

This inconsistency adds difficulty to maintaining a vertical rhythm within a form that includes only some inputs with an assoc. input message.

A quick fix could be to give `<calcite-input-message>` a `display: none` style when its `active` attribute is absent.

**Comment by benelan** [comment]
I'm still able to reproduce on `next`, none of the changes effected the repro sample:
https://codepen.io/benesri/pen/oNyrqLY

**Comment by alisonailea** [comment]
given the implementation of the global hidden attribute, it seems like this doesn't require a code change. Rather the user should apply the "hidden" attribute when the input message should be hidden. A solution which enables an empty input message to be applied in this context without adding any additional space will require a refactor of both label and input message. Is that what we want to do?

**Comment by driskull** [comment]
@eriklharper @macandcheese @benelan wasn't it the plan for this component to go away and be rolled into `input`?

If so, should we deprecate it?

**Comment by macandcheese** [comment]
I know it has been mentioned before, although I can't find an issue for it. I think it might have to be rolled into `label` if possible, since it is applicable to a lot of form components, not just `input`. 

It might be good to re-evaluate this as part of the larger form / validation story. I'd imagine that will be a larger epic for another major release, so not sure if we are quite ready to deprecate it?

**Comment by benelan** [comment]
I honestly thought we were going into the opposite direction by splitting up `input`. If we roll `input-message` into `input` do we just delete `input-text` and `input-number`?

**Comment by macandcheese** [comment]
By rolled into, I just meant having a slot within `label` for it the component to reside. Right now `input-message` as a name is a bit misleading, really it should be / can be used for any form related component messaging, IMO. `form-message` or `label-message`, etc. seems more general / fitting. But I don't know if we need to change this until the next large breaking release after 1.0.

**Comment by benelan** [comment]
Yeah I think that one should wait. Input needs a lot of work; it's all over the place right now. Besides the validation/`input-message` items you mentioned:

- Assuming we want to keep splitting up Input into separate components, we need to
    - start removing `type`s from `calcite-input`. Right now you can do `calcite-input-number` and `calcite-input type="number"`. We need to pick one. †
    - decide if we are creating a component for every `type`, which seems excessive to me. If we are keeping some `type`s in the `calcite-input` component, we need to define which ones stay (and why)
    - create interfaces/utils to cut down on the duplicate code
- Autofill/autocomplete doesn't work well and we will likely need to "unshadow" Input(s). At least until there is better browser support for forms with custom elements
- The lifecycle methods need a refactor

I think it would be better to get all/most of those changes into a single major release if possible.

---
† It may make sense to remove `input-text` and `input-number` before v1 and then add them back when we have time to split the rest of the components, if that's the route we choose. I'll bring that up with Franco when he gets back

**Comment by macandcheese** [comment]
I think that’s a fair summary, although I do think this particular issue scope is much smaller - just fixing the white space with our current component - https://github.com/Esri/calcite-components/pull/6037

**Comment by benelan** [comment]
Yeah I agree that we should fix this white space issue now. But deprecating/moving the `input-message` component should wait until we sort out our Input plans.

**Comment by driskull** [comment]
Adding it as part of label might make the most sense. Or at least a slot of label.

If we do keep it then we should consider renaming it. It's really a form or label message

**Comment by benelan** [comment]
Yeah I like the idea of making it part of label. Input isn't the only form component that can benefit from validation messages

**Comment by driskull** [comment]
Agreed. It should not be tied to just input

**Comment by alisonailea** [comment]
@driskull @benelan the issue with the whitespace "bug" here is two fold. The components involved here use margin which is pushing around content they don't personally have control over (child controlling the layout of it's parent which is a design system anti-pattern) and that the label itself is using flex-gap to simultaneously layout it's children which adds a gap between input and input-message unless input-message's styles are set to "display: none" which can only be achieved by either setting the attribute on the Host or by a user setting the "hidden" attribute themselves. A full fix will require a refactor of label as well as input and input-message. OR documentation instructing users to simply add "hidden" attribute to empty input-message which seems like a better solution at this time.

**Comment by driskull** [comment]
Agreed. Lets hold off on this one until Franco is back. Maybe we can deprecate this component and build it into label after 1.0.

**Comment by alisonailea** [comment]
where should the documentation go?

**Comment by benelan** [comment]
Yeah agreed. I'd put it on the [component's documentation](https://developers.arcgis.com/calcite-design-system/components/input-message/)  page ([source](https://devtopia.esri.com/ArcGISDevelopers/calcite-documentation/blob/dev/documentation/components/input-message.mdx)) under a new Overview/Usage section. Copying the format from other component doc pages, e.g. [alert](https://developers.arcgis.com/calcite-design-system/components/alert/).

@geospatialem do you have the bandwidth to help add the doc?

**Comment by geospatialem** [comment]
> @geospatialem do you have bandwidth to help add the doc?

Definitely, can add it to the doc page today, assuming we're adding a deprecation to the `input-message` component?

**Comment by alisonailea** [comment]
> Definitely, can add it to the doc page today, assuming we're adding a deprecation to the input-message component?

No deprecation yet. But we need to alert users that they must add a "hidden" attribute to an empty input-message or there will be weird extra spacing.

**Comment by geospatialem** [comment]
Added to the doc site, which will go live next Wednesday. Updating to the `Stalled` milestone and Franco's list for consideration for his 1.0 and/or post-1.0 thoughts.

**Comment by jcfranco** [comment]
I think this is no longer valid based on [`input-message`](https://codepen.io/jcfranco/pen/vEyeQvO?editors=1000)'s updated usage guidelines:

> [!NOTE]
> When the component is empty, the hidden property/attribute should be used to hide the component from the DOM, otherwise > additional spacing will be displayed.

[Here's](https://codepen.io/jcfranco/pen/vEyeQvO?editors=1000) the updated repro case.

cc @Esri/calcite-pes

**Comment by DintaMel** [comment]
Closing  per above updated repro shows that using the hidden attribute fixed the issue.
